src/mergesort/libmergesort.c

```diff
@@ -44,9 +43,33 @@
 	typedef uint8_t bool;
 #endif
 
+/* These macros are no longer used in BEEBS. alloca does a better job
+
 #define Var(name, value)				__typeof__(value) name = value
 #define Allocate(type, count)				(type *)malloc((count) * sizeof(type))
 
+*/
+
+
+/* Yield a sequence of random numbers in the range [0, 2^15-1].
+
+   The seed is always initialized to zero.  long int is guaranteed to be at
+   least 32 bits. The seed only ever uses 31 bits (so is positive).
+
+   For BEEBS this gets round different operating systems using different
+   multipliers and offsets and RAND_MAX variations. */
+
+static int
+rand_beebs ()
+{
+  static long int seed = 0;
+
+  seed = (seed * 1103515245L + 12345) & ((1U << 31) - 1);
```

Comment:
Should this be `1UL` to be compatible with 16-bit architectures?

Reply:
Yes. Will fix. Thanks for catching.

Reply:
Fixed in commit 1c9ee79